src/mlpack/tests/loss_functions_test.cpp

```diff
@@ -711,4 +712,65 @@ BOOST_AUTO_TEST_CASE(MarginRankingLossTest)
       "-0.753830 1.336900 0.000000 0.000000 -0.207000 0.328810"), 1e-6);
 }
 
+/**
+ * Simple test for the multiLabel softmargin loss function.
+ */
+BOOST_AUTO_TEST_CASE(MultiLabelSoftMarginLossSumReductionTest)
+{
+  arma::mat input, target, output, weight, expectedOutput;
+  double loss;
+  weight.ones(1, 3);
+  MultiLabelSoftMarginLoss<> module(weight, 3, true);
+
+  input = arma::mat("0.1778 0.1203 -0.2264 0.0957 0.2403 -0.3400 0.1397 0.1925 "
+      "-0.3336");
+  target = arma::mat("0 1 0 1 0 0 0 0 1");
+  expectedOutput = arma::mat("0.1814 -0.1567 0.1479 -0.1587 0.1866 0.1386 "
+      "0.1783 0.1827 -0.1942");
+  input.reshape(3, 3);
+  target.reshape(3, 3);
+  expectedOutput.reshape(3, 3);
+
+  // Test the Forward function. Loss should be 2.14829.
+  loss = module.Forward(input, target);
+  BOOST_REQUIRE_CLOSE(loss, 2.14829, 1e-3);
```

Comment:
Is it manually calculated or by using any framework? If manually then please mention it as a comment, else you can leave it as it is. Same for all the other values.

Reply:
Yes. I have provided a link to a notebook to verify all the calculations in the first comment in this thread. All the values mentioned in any of the test cases can be checked from that notebook. The framework in question is PyTorch.

The notebook has the PyTorch implementation and also an armadillo implementation to reproduce these results. The outputs are also provided along with the implementations.

Reply:
Oh. Didn't notice the ipynb notebook. Thanks :)

Reply:
:)